File: coding_interview_pattern/two_pointers/python/remove_duplicate.py
#!/usr/bin/python3
"""remove occurence of value in array
then return the new length
"""
from typing import List

def removeDuplicate(arr: List[int], n: int) -> int:

    if n == 0 or n == 1: return n

    newLen = 0

    for i in range(n-1):
        if arr[i] != arr[i+1]:
            arr[newLen] = arr[i]
            newLen += 1
    arr[newLen] = arr[n-1]
    newLen += 1

    return newLen



if __name__ == "__main__":
    arr = [1,1,1,3,3,5,5]
    size = len(arr)

    print(arr)

    res = removeDuplicate(arr, size)
    print(res)

    print(arr[:res])

File: coding_interview_pattern/two_pointers/c/remove_duplicate.c
#include <stdio.h>

/**
 * rmDuplicates - remove duplicate in an array
 *
 * @arr: array of sorted duplicate element
 * @n: size of array
 * Return: new length
 */
int rmDuplicates(int arr[], int n)
{
    if (n <= 1) return n;

    int i, newlen = 0;

    for (i = 0; i < n-1; i++)
    {
        if (arr[i] != arr[i+1])
        {
            arr[newlen++] = arr[i];
        }
    }
    arr[newlen++] = arr[n-1];

    return newlen;
}



int main(void)
{
    int arr[] = {1,1,1,3,3,5,5};
    int size = sizeof(arr) / sizeof(arr[0]);

    int newLen = rmDuplicates(arr, size);

    printf("%d\n", newLen);

    for (int i = 0; i < newLen; i++)
        printf("%d ", arr[i]);
    printf("\n");

    return (0);
}
